### DISCUSSION TITLE
Switching to GPU causes

### DISCUSSION THREAD

**Discussion by alarmatwork** [codeproject/CodeProject.AI-Server#239] [Q&A] [answered]
Can anybody advise which _NVIDIA GPU Computing Toolkit_ goes together with the Module 'License Plate Reader' 3.3.4 (ID: ALPR) and CUDDN ver 9.6

I've tried the latest 12.6 and then tried to downgrade to 11.8 versions, but it does not help, still getting:
```
13:06:20:Started License Plate Reader module
13:06:25:ALPR_adapter.py: W1219 13:06:25.980441  6796 dynamic_loader.cc:285] Note: [Recommend] copy cudnn into CUDA installation directory. 
13:06:25:ALPR_adapter.py:  For instance, download cudnn-10.0-windows10-x64-v7.6.5.32.zip from NVIDIA's official website, 
13:06:25:ALPR_adapter.py: then, unzip it and copy it into C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0
13:06:25:ALPR_adapter.py: An exception occurred initialising the module: (PreconditionNotMet) The third-party dynamic library (cudnn64_8.dll) that Paddle depends on is not configured correctly. (error code is 126)
13:06:25:ALPR_adapter.py:   Suggestions:
13:06:25:ALPR_adapter.py:   1. Check if the third-party dynamic library (e.g. CUDA, CUDNN) is installed correctly and its version is matched with paddlepaddle you installed.
13:06:25:ALPR_adapter.py:   2. Configure third-party dynamic library environment variables as follows:
13:06:25:ALPR_adapter.py:   - Linux: set LD_LIBRARY_PATH by `export LD_LIBRARY_PATH=...`
13:06:25:ALPR_adapter.py:   - Windows: set PATH by `set PATH=XXX; (at ..\paddle\phi\backends\dynload\dynamic_loader.cc:312)
13:06:25:ALPR_adapter.py: You should do this according to your CUDA installation directory and CUDNN version.
```

On the Windows machine I should just add proper version toolkit installation dir  _bin_ and _libnvvc_ and CUDNN/bin
![image](https://github.com/user-attachments/assets/a03d83cb-5904-4262-bcf6-d36b49475be7)

**Comment by ChrisMaunder** [comment, marked as answer]
[This script](https://github.com/codeproject/CodeProject.AI-Server/blob/main/devops/install/install_cuDNN.bat) should set things up correctly

**Reply by alarmatwork** [reply]
Thanks a lot. After modifying this script ( Run as Admin detection didnt seem to work on Win11) and removing existing (too new) CUDNN installation.. I got it successfully running on CUDNN version 8.9 by the bat file. Thanks a lot!
